monkey/monkey_island/cc/ui/src/components/pages/RegisterPage.js

```diff
@@ -0,0 +1,123 @@
+
+import AuthService from '../../services/AuthService';
+import {particleParams} from '../../styles/particle-component/RegistrationPageParams';
+import monkeyDetective from '../../images/detective-monkey.svg';
+
+class RegisterPageComponent extends React.Component {
+
+  NO_AUTH_API_ENDPOINT = '/api/environment';
+
+  register = () => {
+    this.auth.register(this.username, this.password).then(res => {
+      this.setState({failed: false, error: ''});
+      if (res['result']) {
+        this.redirectToHome();
+      } else {
+        this.setState({
+          failed: true,
+          error: res['error']
+        });
+      }
+    });
+  };
+
+  setNoAuth = () => {
+    let options = {}
+    options['headers'] = {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    };
+    options['method'] = 'PATCH'
+    options['body'] = JSON.stringify({'server_config': 'standard'})
+
+    return fetch(this.NO_AUTH_API_ENDPOINT, options)
+      .then(res => {
+        if (res.status === 200) {
+          this.auth.attemptNoAuthLogin().then(() => {
+            this.redirectToHome();
+          });
+        }
+        this.setState({
+          failed: true,
+          error: res['error']
+        });
+      })
+  }
+
+  updateUsername = (evt) => {
+    this.username = evt.target.value;
+  };
+
+  updatePassword = (evt) => {
+    this.password = evt.target.value;
+  };
+
+  redirectToHome = () => {
+    window.location.href = '/';
+  };
+
+  constructor(props) {
+    super(props);
+    this.username = '';
+    this.password = '';
+    this.auth = new AuthService();
+    this.state = {
+      failed: false,
+      loading: false
+    };
+
+    this.auth.needsRegistration()
+      .then(result => {
+        if (!result) {
+          this.redirectToHome()
+        }
+      })
+  }
+
+  render() {
+    return (
+      <Container fluid className={'auth-container'}>
+        <Particles className={'particle-background'} params={particleParams}/>
+        <Row>
+          <Col xs={12} lg={{span: 6, offset: 3}} md={{span: 7, offset: 3}}
+               className={'auth-block'}>
+            <Row>
+              <Col lg={8} md={8} sm={8}>
+                <h1 className='reg-title'>First time?</h1>
+                <h3 className='reg-subtitle'>Let's secure your island!</h3>
+                <div>
+                  <Form className={'auth-form'}>
+                    <Form.Control onChange={evt => this.updateUsername(evt)} type='text' placeholder='Username'/>
+                    <Form.Control onChange={evt => this.updatePassword(evt)} type='password' placeholder='Password'/>
+                    <Button id={'auth-button'} onClick={() => {
+                      this.register()
+                    }}>
+                      Lets Go!
+                    </Button>
+                    <a href='#' onClick={this.setNoAuth} className={'no-auth-link'}>
+                      I want anyone to access the island
```

Comment:
perhaps change this to a button as well

Reply:
Nahh, this should be a link with small text that no one clicks :)